docs/ovs-offload.md

```diff
@@ -0,0 +1,199 @@
+
+```
+ip link set enp3s0f0 up
+```
+
+Unbind the VFs from the driver
+
+```
+echo 0000:03:00.2 > /sys/bus/pci/drivers/mlx5_core/unbind
+echo 0000:03:00.3 > /sys/bus/pci/drivers/mlx5_core/unbind
+echo 0000:03:00.4 > /sys/bus/pci/drivers/mlx5_core/unbind
+echo 0000:03:00.5 > /sys/bus/pci/drivers/mlx5_core/unbind
+```
+
+Configure SR-IOV VFs to switchdev mode
+
+```
+devlink dev eswitch set pci/0000:03:00.0 mode switchdev
+ethtool -K enp3s0f0 hw-tc-offload on
+```
+
+Bind the VFs to the driver
+
+```
+echo 0000:03:00.2 > /sys/bus/pci/drivers/mlx5_core/bind
+echo 0000:03:00.3 > /sys/bus/pci/drivers/mlx5_core/bind
+echo 0000:03:00.4 > /sys/bus/pci/drivers/mlx5_core/bind
+echo 0000:03:00.5 > /sys/bus/pci/drivers/mlx5_core/bind
+```
+
+## SR-IOV network device plugin configuration
+
+Create a ConfigMap that defines SR-IOV resource pool configuration
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: sriovdp-config
+  namespace: kube-system
+data:
+  config.json: |
+    {
+      "resourceList": [{
+          "resourcePrefix": "mellanox.com",
+          "resourceName": "cx5_sriov_switchdev",
+          "isRdma": true,
+          "selectors": {
+                  "vendors": ["15b3"],
+                  "devices": ["1018"],
+                  "drivers": ["mlx5_core"]
+              }
+      }
+      ]
+    }
+```
+
+Deploy SR-IOV network device plugin as DaemonSet. See https://github.com/intel/sriov-network-device-plugin.
+
+Deploy multus CNI as DaemonSet. See https://github.com/intel/multus-cni.
+
+Create NetworkAttachementDefinition CRD with Antrea CNI config.
+
+```yaml
+Kubernetes Network CRD Spec:
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: default
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: mellanox.com/cx5_sriov_switchdev
+spec:
+  config: '{
+    "cniVersion": "0.3.1",
+    "name": "antrea",
+    "plugins": [ { "type": "antrea", "ipam": { "type": "host-local" } }, { "type": "portmap", "capabilities": {"portMappings": true} }]
+}'
+```
+## Deploy Antrea Image with hw-offload enabled
+Modify the build/yamls/antrea.yml with offload flag
+
+```yaml
+  - command:
+    - start_ovs
+    - --hw-offload
+```
+
+## Deploy POD with OVS hardware-offload
+
+Create POD spec and
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: ovs-offload-pod1
+  annotations:
+    v1.multus-cni.io/default-network: default
+spec:
+  containers:
+  - name: networkstatic/iperf3
+    image: centos/tools
+    resources:
+      requests:
+        mellanox.com/cx5_sriov_switchdev: '1'
+      limits:
+        mellanox.com/cx5_sriov_switchdev: '1'
+```
+
+## Verify Hardware-Offloads is Working
+
+Run iperf3 server on POD 1
+```
+kubectl exec -it ovs-offload-pod1 -- iperf3 -s
+```
+Run iperf3 clien on POD 2
+
+```
+kubectl exec -it ovs-offload-pod2 -- iperf3 -c 192.168.1.17 -t 100
+```
+
+Check traffic on the VF representor port. Verify only TCP connection establishment appears
+```
+tcpdump -nnn -i eth0
```

Comment:
I may be confused but if this is the "VF representor port", is the name actually "eth0"? I thought this was the interface which is in the host network namespace and is connected to the bridge (with the name is auto-generated by Antera that includes the beginning of the Pod's name)?